crates/viona-api/src/lib.rs

```diff
@@ -19,6 +19,19 @@ impl VionaFd {
 
         let mut vna_create = vioc_create { c_linkid: link_id, c_vmfd: vm_fd };
         let _ = unsafe { this.ioctl(ioctls::VNA_IOC_CREATE, &mut vna_create) }?;
+        #[cfg(feature = "falcon")]
+        if this
+            .ioctl_usize(
+                ioctls::VNA_IOC_SET_PROMISC,
+                vioc_promisc::VIONA_PROMISC_ALL_VLAN as usize,
+            )
+            .is_err()
+        {
+            // Until/unless this support is integrated into stlouis/illumos,
+            // this is an expected failure.   This is needed to use vlans,
+            // but shouldn't affect any other use case.
+            println!("failed to enable promisc mode on viona");
+        }
```

Comment:
Doing this as a feature-gated compile-time decision seems inappropriate to me.  Why not make it a parameter to `PciVirtioViona::new()` and optionally perform the ioctl there?

Reply:
There are plausible reasons why a non-falcon consumer might want to set such a promisc mode, even though there are none presently.  I'd prefer having that painted out through the interface rather than the viona-api crate changing its implicit behavior due to feature presence/absence.

Reply:
It isn't even available in `stlouis` yet.

For context: This new ioctl allows the `viona` driver to receive all vlan-tagged packets, expanding on the existing `VIONA_PROMISC_ALL` ioctl.  I imagine this could be integrated to upstream `illumos` without objection.  The problem is that there is no corresponding mechanism to allow `propolis` to indicate that the driver should transmit all vlan-tagged packets.  It's hard-coded into the driver.  Until I come up with an integrate-able method for handling the `tx` path, `falcon` has to use a non-standard `viona` driver, which it will install as part of its setup.

Reply:
Hmm, that's unfortunate.  While `VIONA_PROMISC_ALL_VLAN` remains in "private" forks, it should probably live behind a `falcon` cfg guard.  Without that, the header checks would otherwise fail on vanilla bits.

With that said, I would still keep the ioctl call itself in propolis-lib, rather than as part of the `VionaFd` constructor.  The viona-api crate is mostly meant as a thin convenience shim over the FFI bits.

Reply:
I've moved the call upstack to `PciVirtioViona::new()` and rebased.

Reply:
@pfmooney, would you mind taking another look at this when you have time?  Thanks.